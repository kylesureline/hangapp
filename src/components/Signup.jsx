import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { firebase } from '../firebase/firebase';
import { startSignup } from '../actions/auth';

export class Signup extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      error: ''
    };
  }
  onInputChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };
  onSubmit = (e) => {
    e.preventDefault();
    const { email, password } = this.state;
    this.props.startSignup(email, password).catch((error) => {
      this.setState(({ error }));
    });
  };
  render() {
    return (
      <form className="form form--login" onSubmit={this.onSubmit}>
        {!!this.state.error.message && <p className="form__error">{
          this.state.error.message.includes('email') ? (
            'Please enter a valid email.'
          ) : (
            'Please enter a password.'
          )
        }</p>}
        <input
          className="text-input"
          type="email"
          name="email"
          placeholder="Email"
          autoFocus
          value={this.state.email}
          onChange={this.onInputChange}
        />
        <input
          className="text-input"
          type="password"
          name="password"
          placeholder="Password"
          value={this.state.password}
          onChange={this.onInputChange}
        />
        <div className="input-group input-group--centered-column">
          <button className="button input-group__item input-group__item--column">Signup</button>
          <Link className="button button--secondary input-group__item input-group__item--column" to={'/'}>Back</Link>
        </div>
      </form>
    );
  }
};

const mapDispatchToProps = (dispatch) => ({
  startSignup: (email, password) => dispatch(startSignup(email, password))
})

export default connect(undefined, mapDispatchToProps)(Signup);
