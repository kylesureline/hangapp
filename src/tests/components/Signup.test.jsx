import React from 'react';
import { shallow } from 'enzyme';
import { Signup } from '../../components/Signup.jsx';

test('should render Signup component correctly', () => {
  const wrapper = shallow(<Signup />);
  expect(wrapper).toMatchSnapshot();
});
