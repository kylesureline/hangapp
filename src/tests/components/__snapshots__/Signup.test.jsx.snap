// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Signup component correctly 1`] = `
<form
  className="form form--login"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="email"
    value=""
  />
  <input
    className="text-input"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <div
    className="input-group input-group--centered-column"
  >
    <button
      className="button input-group__item input-group__item--column"
    >
      Signup
    </button>
    <Link
      className="button button--secondary input-group__item input-group__item--column"
      to="/"
    >
      Back
    </Link>
  </div>
</form>
`;
